Net evaluation file missing in build dependencies for `pgo` target

`make pgo` on a freshly cloned (or cleaned) repository does not compile:

```
$ git clean -dfx
Removing .tmp/
Removing Quanticade.d
Removing net44.nnue
$ make pgo
cc -g -std=gnu11 -funroll-loops -O3 -flto -fno-exceptions -DIS_64BIT -DNDEBUG -Wall -Werror -Wextra -Wno-error=vla -Wpedantic -Wno-unused-command-line-argument -DUSE_AVX2 -DUSE_SIMD -mavx2 -mbmi -mbmi2 -DNETWORK_NAME=\"net44.nnue\" -DEVALFILE=\"net44.nnue\"  -march=native  -MMD -MP -o Quanticade Source/attacks.c Source/datagen.c Source/evaluate.c Source/history.c Source/move.c Source/movegen.c Source/nnue.c Source/perft.c Source/quanticade.c Source/search.c Source/see.c Source/spsa.c Source/stats.c Source/syzygy.c Source/threads.c Source/transposition.c Source/uci.c Source/utils.c  -lm 
lto-wrapper: warning: using serial compilation of 2 LTRANS jobs
lto-wrapper: note: see the '-flto' option documentation for more information
/tmp/ccoN16xo.s: Assembler messages:
/tmp/ccoN16xo.s:11: Error: file not found: net44.nnue
lto-wrapper: fatal error: cc returned 1 exit status
compilation terminated.
/usr/bin/x86_64-linux-gnu-ld.bfd: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
make: *** [Makefile:268: pgo] Error 1
```

The following change in the makefile should fix this issue:
```
diff --git a/Makefile b/Makefile
index 7a1ffb1..50f2287 100644
--- a/Makefile
+++ b/Makefile
@@ -264,7 +264,7 @@ $(TMPDIR):
 
 
 # Usual disservin yoink for makefile related stuff
-pgo:
+pgo: $(EVALFILE)
        $(CC) $(CFLAGS) $(PGO_GEN) $(NATIVE) $(INSTRUCTIONS) -MMD -MP -o $(EXE) $(SOURCES) -lm $(LDFLAGS)
        ./$(EXE) bench
        $(PGO_MERGE)
```

--- a/Makefile
+++ b/Makefile
@@ -264,7 +264,7 @@ $(TMPDIR):
 
 
 # Usual disservin yoink for makefile related stuff
-pgo:
+pgo: $(EVALFILE)
 	$(CC) $(CFLAGS) $(PGO_GEN) $(NATIVE) $(INSTRUCTIONS) -MMD -MP -o $(EXE) $(SOURCES) -lm $(LDFLAGS)
 	./$(EXE) bench
 	$(PGO_MERGE)

